File: src/main/java/team4/footwithme/team/service/response/TeamDefaultResponse.java
package team4.footwithme.team.service.response;

import team4.footwithme.team.domain.Team;

public record TeamDefaultResponse(
        Long teamId,
        Long stadiumId,
        String name,
        String description,
        int winCount,
        int drawCount,
        int loseCount,
        String location
) {
        public static TeamDefaultResponse from(Team team) {
                return new TeamDefaultResponse(
                        team.getTeamId(),
                        team.getStadiumId(),
                        team.getName(),
                        team.getDescription(),
                        team.getTotalRecord().getWinCount(),
                        team.getTotalRecord().getDrawCount(),
                        team.getTotalRecord().getLoseCount(),
                        team.getLocation()
                );
        }

}

File: src/main/java/team4/footwithme/team/service/response/TeamDefaultResponse.kt
package team4.footwithme.team.service.response

import team4.footwithme.team.domain.Team

@JvmRecord
data class TeamDefaultResponse(
    val teamId: Long?,
    val stadiumId: Long?,
    @JvmField val name: String?,
    @JvmField val description: String?,
    val winCount: Int,
    val drawCount: Int,
    val loseCount: Int,
    @JvmField val location: String?
) {
    companion object {
        fun from(team: Team): TeamDefaultResponse {
            return TeamDefaultResponse(
                team.teamId,
                team.stadiumId,
                team.name,
                team.description,
                team.totalRecord!!.winCount,
                team.totalRecord!!.drawCount,
                team.totalRecord!!.loseCount,
                team.location
            )
        }
    }
}
